crates/flox-rust-sdk/src/providers/git.rs

```diff
@@ -432,15 +445,80 @@ impl GitProvider for GitCommandProvider {
         let _out = GitCommandProvider::run_command(
             GitCommandProvider::new_command(&self.workdir)
                 .arg("branch")
-                .arg(branch)
-                .arg("--set-upstream")
+                .arg("--set-upstream-to")
                 .arg(format!("{origin_name}/{branch}")),
         )
         .await?;
 
         Ok(())
     }
 
+    /// Retrieve information about the remot origin for the current branch/repo
+    ///
+    /// Return a tuple containing
+    ///
+    /// 1. the remote name of the current branch (or "origin" if no upstream configured)
+    /// 2. the remote url
+    /// 3. (if configured) a tuple containing
+    ///    1. the upstream branch name
+    ///    2. the current revision of the branch
+    async fn get_origin(
+        &self,
+    ) -> Result<(String, String, Option<(String, String)>), Self::GetOriginError> {
+        let (remote_name, remote_branch) = {
+            let reference = GitCommandProvider::run_command(
+                GitCommandProvider::new_command(&self.workdir)
+                    .arg("rev-parse")
+                    .arg("--abbrev-ref")
+                    .arg("--symbolic-full-name")
+                    .arg("@{u}"),
+            )
+            .await;
+
+            match reference {
+                Err(_) => {
+                    error!("Couldn't determine upstream remote name for the current branch, defaulting to 'origin'");
```

Comment:
I think you're handling it correctly, but it's also possible the local repo is not on a branch. Might be worth reflecting that in comments or this error

Reply:
Yeah I think `HEAD` is more correct.

The function is kind of confusing method right now, because it can have two completely different behaviors - getting the upstream branch for the current branch, or getting origin. I'm not sure what the use case for the latter would be; your current code just errors in that case, so I wonder if it's necessary.

Reply:
I think we used to try to be more helpful before by guessing at least the upstream repo even if there is none configured for the current branch..
I'm happy to just cause an error in that case at the git level